src/rules/noMagicNumbersRule.ts

```diff
@@ -63,44 +61,31 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static DEFAULT_ALLOWED = [ -1, 0, 1 ];
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NoMagicNumbersWalker(sourceFile, this.getOptions()));
+        const options = this.getOptions();
+        const ruleArguments = options.ruleArguments;
+        const allowedNumbers = ruleArguments.length > 0 ? ruleArguments : Rule.DEFAULT_ALLOWED;
+        return this.applyWithWalker(new NoMagicNumbersWalker(sourceFile, options.ruleName, new Set(allowedNumbers.map(String))));
     }
 }
 
-class NoMagicNumbersWalker extends Lint.RuleWalker {
-    // allowed magic numbers
-    private allowed: Set<string>;
-    constructor(sourceFile: ts.SourceFile, options: IOptions) {
-        super(sourceFile, options);
-
-        const configOptions = this.getOptions();
-        const allowedNumbers: number[] = configOptions.length > 0 ? configOptions : Rule.DEFAULT_ALLOWED;
-        this.allowed = new Set(allowedNumbers.map(String));
-    }
-
-    public visitNode(node: ts.Node) {
-        const num = getLiteralNumber(node);
-        if (num !== undefined) {
-            if (!Rule.ALLOWED_NODES.has(node.parent!.kind) && !this.allowed.has(num)) {
-                this.addFailureAtNode(node, Rule.FAILURE_STRING);
+class NoMagicNumbersWalker extends Lint.AbstractWalker<Set<string>> {
+    public walk(sourceFile: ts.SourceFile) {
+        const cb = (node: ts.Node): void => {
+            if (node.kind === ts.SyntaxKind.NumericLiteral) {
+                this.checkNumericLiteral(node, (node as ts.NumericLiteral).text);
+            } else if (node.kind === ts.SyntaxKind.PrefixUnaryExpression &&
```

Comment:
I agree, but I don't think it's a good idea to scatter the declarations of typeguard functions through the whole codebase. Currently it is done this way which leads to many duplicated small functions. Maybe we should add them to a new module `src/language/typeguard.ts` as needed.

But that's not directly related to this PR and can be considered when migrating other exisiting rules.